fast_llm/layers/attention/attention.py

```diff
@@ -429,3 +436,115 @@ def get_compute_usage(self, input_: TensorMeta, kwargs: dict[str, typing.Any], c
                 self.dense.get_compute_usage(dense_input, config),
             )
         )
+
+    def preprocess(self, batch: torch.Tensor, kwargs: dict[str, typing.Any]) -> None:
+        self._rotary.preprocess(batch, kwargs)
+        if not self._use_flash_attention:
```

Comment:
could we make this a config option, an enum? we select other implementations like that

Reply:
It's already an option. We could turn into an enum, but I don't see too much reason

Reply:
It would be useful if we could select the varlen version this way. That would need extra work because it's currently determined in `BatchConfig.cross_document_attention` and affects data processing, so won't do now but that's something to consider for the future.